packages/serverless-offline-sqs/src/index.js

```diff
@@ -74,25 +74,27 @@ class ServerlessOfflineSQS {
     serverlessLog(`Starting Offline SQS: ${this.options.stage}/${this.options.region}.`);
   }
 
-  async ready() {
+  ready() {
     if (process.env.NODE_ENV !== 'test') {
-      await this._listenForTermination();
+      this._listenForTermination();
     }
   }
 
-  // eslint-disable-next-line class-methods-use-this
-  async _listenForTermination() {
-    const command = await new Promise(resolve => {
-      process.on('SIGINT', () => resolve('SIGINT')).on('SIGTERM', () => resolve('SIGTERM'));
-    });
+  _listenForTermination() {
+    const signals = ['SIGINT', 'SIGTERM'];
+
+    signals.map(signal =>
+      process.on(signal, async () => {
+        serverlessLog(`Got ${signal} signal. Offline Halting...`);
 
-    serverlessLog(`Got ${command} signal. Offline Halting...`);
+        await this.end();
```

Comment:
Does process.on callback ever support async functions?

Reply:
Yup, run the script in the shell in the following repl.it: https://replit.com/@larrybotha1/processon-async#index.js